pallets/ema-oracle/src/lib.rs

```diff
@@ -232,7 +247,43 @@ pub mod pallet {
 	}
 
 	#[pallet::call]
-	impl<T: Config> Pallet<T> {}
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::add_oracle())]
+		pub fn add_oracle(origin: OriginFor<T>, source: Source, assets: (AssetId, AssetId)) -> DispatchResult {
+			T::AuthorityOrigin::ensure_origin(origin)?;
+
+			let assets = ordered_pair(assets.0, assets.1);
+
+			WhitelistedAssets::<T>::mutate(|list| {
+				list.try_insert((source, (assets)))
+					.map_err(|_| Error::<T>::TooManyUniqueEntries)
+			})?;
+
+			Self::deposit_event(Event::OracleAdded { source, assets });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(<T as Config>::WeightInfo::remove_oracle())]
+		pub fn remove_oracle(origin: OriginFor<T>, source: Source, assets: (AssetId, AssetId)) -> DispatchResult {
```

Comment:
i would expect this function would remove actual oracle as well, not just item from whitelist